lib/dependency-manager-adapters/npm.js

```diff
@@ -102,11 +101,7 @@ module.exports = CoreObject.extend({
       mgrOptions = mgrOptions.concat(['--no-shrinkwrap']);
     }
 
-    return this.run(cmd, [].concat(['install'], mgrOptions), { cwd: this.cwd }).then(() => {
-      if (!adapter.useYarnCommand) {
-        return adapter.run(adapter.configKey, ['prune'], { cwd: adapter.cwd });
```

Comment:
@mansona - Should we keep this around when using npm < 5?

Reply:
The addon blueprint still is node 6 + npm 4 I believe.

Reply:
regarding the addon blueprint https://github.com/ember-cli/ember-cli/pull/8426 😏 

but to be honest I don't actually know the benefit of keeping it around for npm@4 🤔 npm@4 is such an antiquated technology that I think we should be discouraging the use of it as much as possible. Also, It's worth noting that it was released pre yarn@1.0 🙈 and I'm assuming we don't support yarn 0.28 which is the latest version from when npm@4 was last released

Reply:
We would have to major bump here because of the existing addons testing with npm 4.

Reply:
I'm 👍 with that. Integers are cheap 🎉 

Also I think this is really quite an important change because anyone trying this with the default npm on Node 8 or Node 10 will have failing tests for their addons 😞 It's not a great DX and I think a major bump is a tiny pain compared to the issues that this is causing

Reply:
Re ^ -- It's not always true. I haven't figured out specifically what causes this problem. I am certainly using the default `npm` on newer Nodes in my day-to-day and _only_ just recently encountered/recreated this problem myself with that one repo I linked you.

Reply:
If we inspect the version of node and conditionally do this, we can release it as a bugfix and those for whom this is broken (or would be -- depending on the blueprint they start with) will get the fix without having to search out and manually upgrade ember-try.

Reply:
So the specific **cause** of this problem is that when you run `npm prune` and you have something installed that isn't available in the `package-lock.json` it instantly gets deleted. 

I'm going to try this fix again in a way that will actually show the issue and work in npm@4 and npm@5 👍